token/program2/src/instruction.rs

```diff
@@ -196,14 +220,58 @@ impl TokenInstruction {
         }
         Ok(match input[0] {
```

Comment:
@jackcmay - all this `unsafe` is so icky.  There's really no other way?

Reply:
I went with the most efficient.  Even when I tried a hand written ser/de, the number of overhead went up considerably.  I think the right answer here is to jump into the compiler/linker/intrinsics and optimize that stuff.